src/db/models.rs

```diff
@@ -173,31 +179,59 @@ impl DBManager {
     }
 
     pub fn get_bso(&self, collection_id: i64, bso_id: &str) -> Result<Option<BSO>, DieselError> {
-        let result = bso::table
+        bso::table
             .select(bso::table::all_columns())
             .filter(bso::collection_id.eq(&collection_id))
             .filter(bso::id.eq(&bso_id))
             .filter(bso::expiry.ge(&ms_since_epoch()))
-            .first::<BSO>(&self.conn);
-        match result {
-            Ok(bso) => Ok(Some(bso)),
-            Err(NotFound) => Ok(None),
-            Err(e) => Err(e),
-        }
+            .first::<BSO>(&self.conn)
+            .optional()
+    }
+
+    pub fn get_bso_modified(&self, collection_id: i64, bso_id: &str) -> Result<i64, DieselError> {
+        let last_modified = bso::table
+            .select(bso::last_modified)
+            .filter(bso::collection_id.eq(&collection_id))
+            .filter(bso::id.eq(&bso_id))
+            .filter(bso::expiry.ge(&ms_since_epoch()))
+            .first(&self.conn)
+            .optional()?;
+        // XXX: returning 0 when not found or prefer Option?
```

Comment:
FWIW, my gut feeling is it'll be better off returning an `Option`, but my rust-related gut feelings are not exactly well developed yet...

Reply:
In general I agree (it's an Option) but I suspect 0 might work ok for how the view layer uses this. Left this as a note to revisit after we implement those views